lib/grpc/client/interceptors/logger.ex

```diff
@@ -14,11 +14,11 @@ defmodule GRPC.Client.Interceptors.Logger do
   ## Usage
 
       {:ok, channel} = GRPC.Stub.connect("localhost:50051", interceptors: [GRPC.Client.Interceptors.Logger])
-      # This will log on `:info` and greater priority
+      # This will log on `:info` and lower priority (debug)
```

Comment:
I think we should change the default to be `[:gt, :eq]` to keep with the default Logger behavior

Reply:
That's a valid approach too. My main concern is how the API would read to users.

With the change I did in this PR, the API reads like this:

- `[:lt, :eq]` means it will log on `:info` and lower priority (`lt` = `lower`)
- `[:gt, :eq]` means it will log on `:info` and higher priority (`gt` = `higher`)

Whereas if we just change the default to `[:gt, :eq]` and keep the order on `Logger.compare_levels`, the API reads like:

- `[:gt, :eq]` means it will log on `:info` and lower priority (but `gt` means higher)
- `[:lt, :eq]` means it will log on `:info` and higher priority (but `lt` means higher)

I think the first keeps things more clear. WDYT?

Reply:
I don't mean changing the default _instead_ of your changes.

I mean in addition to them. I don't think it makes sense to have the default logging info+debug instead of info+warn+error+...

Reply:
The default behavior of Logger is `[:lt, :eq]`, not `[:gt, :eq]`.

Example:

```elixir
iex(1)> require Logger
Logger
iex(2)> Logger.configure(level: :debug) # should print all level of logs
:ok
iex(3)> Logger.info("I am being logged because debug < info")

[info]  I am being logged because debug < info
:ok
iex(4)> Logger.configure(level: :warn) 
:ok
iex(5)> Logger.info("I am not being logged because warn > info")
:ok
```

If we do `[:gt, :eq]` and I have my application log level set to `debug`, elixir-grpc logs (that use `:info` by default) won't be printed, not matching Logger's behavior. That's the issue described on https://github.com/elixir-grpc/grpc/issues/325

Reply:
> I might have mixed up the comparators, but the point that we should change the default so that by passing info you log everything but debug still stands.

I think I'm missing something here.

Here's my thought process: The interceptor `level` configuration determines what log level the interceptor itself will use to log the requests coming in and out. If you set it to `:warn`, `GRPC.Client.Interceptors.Logger` will call `Log.warn(...)`. If you set it to `:debug`, `GRPC.Client.Interceptors.Logger` will call `Log.debug(...)`.

Now, it's up to the `:logger` configuration to determine what level should be printed or not. If you configure `:logger` level to `:info` but the log level of `GRPC.Client.Interceptors.Logger` is `:debug`, then nothing should be logged (as long as default is `[:lt, :eq]`).

If we keep `[:gt, :eq]` as the default using the scenario above (logger level = `:info`, interceptor level = `:debug`), the interceptor will enter the `if Logger.compare_levels` condition and will attempt to log, using resources unnecessarily to calculate the time diffs, while Elixir will discard the log.

I'm missing the use case of `accepted_comparators` because I couldn't understand in which use case it should be customized to anything besides `[:lt, :eq]`. If you don't bother to explain, can you share a real use case of that? 😅

Reply:
> I'm missing the use case of accepted_comparators because I couldn't understand in which use case it should be customized to anything besides [:lt, :eq]. If you don't bother to explain, can you share a real use case of that? 😅

That wasn't something I decided on haha! I just refactored a bit on this code that was already present.

Reply:
tl; dr: I'll make the changes I suggested, and as I see it, we're in overall agreement on how the code should work :)

Reply:
registering what we talked on Slack:

Turns out that the `accepted_comparators` option is too confusing and we don't see an actual application in which the option makes sense. Therefore, we're gonna remove the option and revert to the previous behavior.